bgp/gobgp/plugin_impl_gobgp_test.go

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2017 Pantheon technologies s.r.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//Package gobgp_test contains Ligato GoBGP Plugin implementation tests
+package gobgp_test
+
+import (
+	"errors"
+	"github.com/ligato/bgp-agent/bgp"
+	"github.com/ligato/bgp-agent/bgp/gobgp"
+	"github.com/ligato/cn-infra/core"
+	"github.com/ligato/cn-infra/flavors/local"
+	"github.com/ligato/cn-infra/logging/logroot"
+	"github.com/osrg/gobgp/config"
+	bgpPacket "github.com/osrg/gobgp/packet/bgp"
+	"github.com/osrg/gobgp/server"
+	"github.com/osrg/gobgp/table"
+	"github.com/stretchr/testify/assert"
+	"sync"
+	"testing"
+	"time"
+)
+
+const (
+	nextHop1                string = "10.0.0.1"
+	nextHop2                string = "10.0.0.3"
+	prefix1                 string = "10.0.0.0"
+	prefix2                 string = "10.0.0.2"
+	length                  uint8  = 24
+	expectedReceivedAs             = uint32(65000)
+	maxSessionEstablishment        = 2 * time.Minute
+	timeoutForNotReceiving         = 5 * time.Second
+)
+
+var flavor = &local.FlavorLocal{}
+
+func TestGoBGPPluginInfoPassing(t *testing.T) {
```

Comment:
Please explain in comment what is the intent of this test.

Consider having: Given/When/Then test steps to do the test scenario readable.

Reply:
comment added